modules/sdk-core/src/bitgo/lightning/validateLnurlInvoice.ts

```diff
@@ -0,0 +1,63 @@
+import * as bolt11 from 'bolt11';
+import * as crypto from 'crypto';
+
+export type ParsedLightningInvoice = {
+  amount: number;
+  /** hex-encoded */
+  paymentHash: string;
+  /** hex-encoded */
+  pubkey: string;
+};
+
+function getNetworkForInvoice(invoice: string) {
+  if (invoice.startsWith('lntbs')) {
+    // signet invoices are not supported by our bolt11 dependency
+    // because the `tbs` prefix used for the invoice does not match the `tb`
+    // prefix used by on-chain signet addresses
+    // see: https://github.com/bitcoinjs/bolt11/pull/58#issuecomment-1106495709
+    // we can still decode the invoice however using a custom network
+    return {
+      bech32: 'tbs',
+      pubKeyHash: 0x6f,
+      scriptHash: 0xc4,
+      validWitnessVersions: [0, 1],
+    };
+  }
+
+  return undefined;
+}
+
+export function validateLnurlInvoice(coin: string, invoiceStr: string, milliSatAmount: string, metadata: string): void {
```

Comment:
Missing docstring

It is easy to forget using a validation method. Also it is often unclear when to use it - any time you receive this set of parameters somewhere?

Better add a `parseLnurlInvoice(...): ParsedInvoice` that does the validation and ensures that the resulting type is valid.